### PR TITLE
Refactor migration controller

### PR DESCRIPTION
**Pull request by rmohr**
Use only one loop with informers for pods, to back the loop.

### CODE REVIEW

File: pkg/virt-controller/watch/migration.go [hunk 1]
```diff
@@ -2,59 +2,92 @@ package watch
...
+	informer   cache.ControllerInterface
+}
+
+func (c *MigrationController) Run(threadiness int, stopCh chan struct{}) {
+	defer kubecli.HandlePanic()
+	defer c.queue.ShutDown()
+	logging.DefaultLogger().Info().Msg("Starting controller.")
+
+	// Start all informers and wait for the cache sync
+	_, jobInformer := NewMigrationJobInformer(c.clientset, c.queue)
```
Comment (admiyo) [inline pkg/virt-controller/watch/migration.go]: It is really confusing to call this a MigrationJob.  it Is not a K8S Job.  

There is the pod that runs the migration, and the pod that is the mirgation target.  Suggest that the first be called the MigrationProcessPod and the second the MigrationTargetPod

File: pkg/virt-controller/watch/migration.go [hunk 2]
```diff
@@ -85,16 +111,21 @@ func (md *MigrationDispatch) execute(store cache.Store, key string) error {
...
 
-	var migration *kubev1.Migration = obj.(*kubev1.Migration)
-	logger := logging.DefaultLogger().Object(migration)
+	logger := logging.DefaultLogger().Object(obj.(*kubev1.Migration))
+	// Copy migration for future modifications
+	if obj, err = kubeapi.Scheme.Copy(obj.(runtime.Object)); err != nil {
+		logger.Error().Reason(err).Msg("could not copy migration object")
+		return err
+	}
+	migration := obj.(*kubev1.Migration)
```
Comment (admiyo) [inline pkg/virt-controller/watch/migration.go]: The current bundling of object creation functions together is going to make it harder to share the caches and replace REST calls like this with cache lookups.

Comment (admiyo) [inline pkg/virt-controller/watch/migration.go]: Nothing tests case kubev1.MigrationScheduled:  which is probably the biggest missing piece

Comment (rmohr) [inline pkg/virt-controller/watch/migration.go]: Regarding the "bundling", I can't follow you ...

Comment (rmohr) [inline pkg/virt-controller/watch/migration.go]: MigrationScheduled is removed. It is now neede anymore, now that everything is in one controller loop

**Comment by rmohr** [comment]
> Net loss of tests. What happened to the tests from test_job.go?

They were testing stuff, which we don't need anymore, now that we only have one controller loop. I will add more tests, which test error scenarios, which were not tested before ...

**Comment by admiyo** [comment]
This needs a  rebase.  Changes seem to be minor, but non-trivial.

**Comment by rmohr** [comment]
@admiyo rebased. All tests pass locally

**Pull request merged**